setup/install_gatling/20_foss_install/10_install_dev_repos.rb

```diff
@@ -0,0 +1,184 @@
+
+    # Scrape the puppet-agent repo page for available puppet-agent builds and
+    # filter down to only released builds (e.g., 1.2.3) vs.
+    # some mergely/SHA version.  The list of builds is ordered from most recent to
+    # oldest.  The resulting list is passed along to the get_cent7_repo routine,
+    # which returns a URL for the first build which has a cent7 repo in it.
+    opts[:puppet_agent_version] = get_cent7_repo(
+        response.lines.
+            select { |l| l =~ /<td><a / }.
+            select { |l| l.match(/^.*href="(\d+\.\d+\.\d+)\/\?C=M&amp;O=D".*$/) },
+        "puppet-agent")
+  end
+
+  # Mostly copied from: install_puppet_agent_dev_repo_on in Beaker's foss_utils.rb
+  def install_puppet_agent_dev_repo_from_local_file( host, opts )
+    opts[:copy_base_local]    ||= File.join('tmp', 'repo_configs')
+    opts = FOSS_DEFAULT_DOWNLOAD_URLS.merge(opts)
+
+    variant, version, arch, codename = host['platform'].to_array
+    add_role(host, 'aio') #we are installing agent, so we want aio role
+    copy_dir_local = File.join(opts[:copy_base_local], variant)
+    onhost_copy_base = opts[:copy_dir_external] || host.external_copy_base
+
+    release_path_end, release_file = host.puppet_agent_dev_package_info(
+        "PC1", opts[:puppet_agent_version], opts )
+    release_path = "#{opts[:dev_builds_url]}/puppet-agent/#{ opts[:puppet_agent_commit] }/repos/"
+    release_path << release_path_end
+    logger.trace("#install_puppet_agent_dev_repo_on: dev_package_info, continuing...")
+
+    if variant =~ /eos/
+      host.get_remote_file( "#{release_path}/#{release_file}" )
+    else
+      onhost_copied_file = File.join(onhost_copy_base, release_file)
+      fetch_http_file( release_path, release_file, copy_dir_local)
+      scp_to host, File.join(copy_dir_local, release_file), onhost_copy_base
+    end
+
+    case variant
+      when /eos/
+        host.install_from_file( release_file )
+      when /^(sles|aix|fedora|el|centos)$/
+        # NOTE: AIX does not support repo management. This block assumes
+        # that the desired rpm has been mirrored to the 'repos' location.
+        # NOTE: the AIX 7.1 package will only install on 7.2 with
+        # --ignoreos. This is a bug in package building on AIX 7.1's RPM
+        if variant == "aix" and version == "7.2"
+          aix_72_ignoreos_hack = "--ignoreos"
+        end
+        on host, "rpm -ivh #{aix_72_ignoreos_hack} #{onhost_copied_file}"
+      when /^windows$/
+        result = on host, "echo #{onhost_copied_file}"
+        onhost_copied_file = result.raw_output.chomp
+        msi_opts = { :debug => host[:pe_debug] || opts[:pe_debug] }
+        install_msi_on(host, onhost_copied_file, {}, msi_opts)
+      when /^osx$/
+        host.install_package("puppet-agent-#{opts[:puppet_agent_version]}*")
+      when /^solaris$/
+        host.solaris_install_local_package( release_file, onhost_copy_base )
+    end
```

Comment:
Does this need to throw an error if none of the cases match?

Reply:
It doesn't in Beaker?  This code is temporary.  Once we get the other AWS account with access to our network, we will get rid of most of my custom install code and just go back to using what Beaker does.